File: Sources/BiangBiangUI/Config/SettingDescriptor.swift
public struct SettingDescriptor: Sendable, Identifiable {
    public enum Kind: Sendable { case toggle, picker(options: [String]) }
    public let key: String
    public let kind: Kind
    public let label: String
    public let defaultValue: String        // "true"/"false" for toggles; option for pickers
    public let footer: String?
    public var id: String { key }
    public init(key: String, kind: Kind, label: String,
                defaultValue: String, footer: String? = nil) {
        self.key = key; self.kind = kind; self.label = label
        self.defaultValue = defaultValue; self.footer = footer
    }
}

File: android/biangbiang-ui/src/main/java/dev/veeso/biangbiangui/config/SettingDescriptor.kt
package dev.veeso.biangbiangui.config

/**
 * Declarative descriptor for an app-supplied extra setting.
 * Mirrors iOS `SettingDescriptor`.
 */
data class SettingDescriptor(
    val key: String,
    val kind: Kind,
    val label: String,
    /** "true"/"false" for toggles; option for pickers. */
    val defaultValue: String,
    val footer: String? = null,
) {
    /** Stable identity, mirrors iOS `Identifiable.id`. */
    val id: String get() = key

    /** Mirrors iOS `SettingDescriptor.Kind`. */
    sealed class Kind {
        data object Toggle : Kind()

        data class Picker(val options: List<String>) : Kind()
    }
}
